src/main/java/com/jherrick/BigBoard.java

```diff
@@ -146,20 +140,25 @@ public BoardState getBoardState() {
 	 **/
 	@Override
 	public List<Move> getLegalMoves() {
+		
 		List<Move> moves = new ArrayList<>();
-		// TODO: perhaps this is not the most efficient, combining ArrayLists...
+		Move move = this.getLastMove();
 
-		// get the last move
-		Move m = this.getLastMove();
+		// if first move, generate all moves
+		if (move == null){
+			moves = allPossibleMoves(boards);
+			return moves;
+		}
 
 		// if target SubBoard is IN_PROGRESS, only generate moves in that SubBoard
-		if (m != null && boards[m.translate()].getState() == BoardState.IN_PROGRESS) {
-			moves.addAll(getLegalMoves(m.translate()));
+		
+		if (boards[move.getSubBoardTarget()].getState() == BoardState.IN_PROGRESS) {
```

Comment:
Doesn't have to be in this PR, but statements like this can definitely be extracted into methods for better readability.